File: draymed_js/tests/test_check_master_codes.ts

function hashFile(fileName: string): string {
    const crypto = require('crypto');
    const fs = require('fs');
    const src_file_buffer = fs.readFileSync(fileName);
    const sum = crypto.createHash('sha256');
    sum.update(src_file_buffer);
    return sum.digest('hex');
}

describe('check that local copy of master codes file ', () => {

    test('is identical with the source file', () => {

        const sourceFile = __dirname + "/../../master_codes.json";
        const hashSource = hashFile(sourceFile);
        console.log(hashSource);

        const targetFile = __dirname + "/../src/data/master_codes.json";
        const hashTarget = hashFile(targetFile);
        console.log(hashTarget);

        expect(hashSource).toEqual(hashTarget);
    });

});

File: draymed_py/tests/test_check_master_codes.py
import filecmp
from pathlib import Path


class TestCheckMasterCodes:
    # Check that local copy of master codes file is identical with the source file.
    def test_codes_file_is_consistent(self) -> None:

        source_filepath: Path = (
            Path(__file__).parent.parent.parent / "master_codes.json"
        )

        target_filepath: Path = (
            Path(__file__).parent.parent / "draymed/data/master_codes.json"
        )

        assert filecmp.cmp(source_filepath, target_filepath)
